lib/ui/profile/usecases/fetch_profile_use_case.dart

```diff
@@ -9,23 +10,43 @@ import 'package:hypha_wallet/ui/profile/interactor/profile_data.dart';
 class FetchProfileUseCase {
   final ProfileService _profileService;
   final HyphaSharedPrefs _appSharedPrefs;
+  final DaoService _daoService;
 
-  FetchProfileUseCase(this._profileService, this._appSharedPrefs);
+  FetchProfileUseCase(this._profileService, this._appSharedPrefs, this._daoService);
 
-  Future<Result<ProfileData, HyphaError>> run(String accountName, Network network) async {
-    final Result<ProfileData, HyphaError> result = await _profileService.getProfile(accountName, network);
-    if (result.isValue) {
-      final ProfileData profile = result.asValue!.value;
+  Future<Result<ProfileData, HyphaError>> run(UserProfileData userProfileData) async {
+    final Future daosFuture = _daoService.getDaos(
+      accountName: userProfileData.accountName,
+      network: userProfileData.network,
+    );
+    final profileFuture = _profileService.getProfile(
+      userProfileData.accountName,
+      userProfileData.network,
+    );
+
+    final futureResults = await Future.wait([
```

Comment:
instead of patching the data together, can we have the profile service return this info? 
@n13

Reply:
No. ProfileService is a mess and we need to avoid it as much as possible until it's refactored and beautiful and then we can do things like this very easily as it's all going to be in graph QL @Alex-IS